doc/CodeStyle.md

```diff
@@ -150,3 +150,49 @@ Bad
     int fd;  
         /* ^^ Remove trailing space */
 ```
+
+### macro definition
+
+Good
+```C
+    #define UCS_MACRO_SHORT(_obj, _field, _val) \
+        (_obj)->_field = (_val)
+
+    #define UCS_MACRO_LONG(_obj, _field1, _field2, _val1, _val2) \
+        do { \
```

Comment:
We do not really need  `do while`, because `{}` are required for each statement according to our coding style.

Reply:
Agree, let's add a variable declared in a macro. So, the `{}` is required.
My point was to show that `do {} while` should be used instead of `{}` to avoid compiler warnings about unnecessary `;` after macro

Reply:
I meant that we do not use `if` or `else` without `{}`, so no `;` problems should arise. 
AFAIR, initially we specifically avoided `do while` macro constructions, but now we have some of them thru the code though.

Reply:
let's not encourage do .. while since in UCX we require { } in if/else

Reply:
Can I ask again why we don't automate this with https://zed0.co.uk/clang-format-configurator and https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/clang-format-diff.py

Since we have have azure how, we can make it part of regular check.

Reply:
@shamisp i you have cycles for it - welcome to add

Reply:
@yosefe I will give it a try. I guess I can go with the code format of my choice ....

Reply:
> let's not encourage do .. while since in UCX we require { } in if/else

done